src/components/PeopleList.tsx

```diff
@@ -0,0 +1,95 @@
+import { useParams } from 'react-router-dom';
+import { useEffect, useState } from 'react';
+import { getPeople } from '../api';
+import { Person } from '../types';
+import { Loader } from './Loader';
+import { PersonItem } from './PersonItem';
+
+function getPreparedPeople(people: Person[]) {
+  return people.map(person => {
+    return {
+      ...person,
+      mother: people.find(mother => mother.name === person.motherName),
+      father: people.find(father => father.name === person.fatherName),
+    };
+  });
+}
+
+export const PeopleList = () => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isError, setIsError] = useState(false);
+  const { personId = '' } = useParams();
+
+  useEffect(() => {
+    setIsLoading(true);
+
+    getPeople()
+      .then(data => {
+        setPeople(getPreparedPeople(data));
+      })
+      .catch(() => {
+        setIsError(true);
+      })
+      .finally(() => {
+        setIsLoading(false);
+      });
+  }, []);
+
+  const isPeopleOnServer = !!people.length && !isError;
+  const isNoPeopleOnServer = !people.length && !isLoading;
+
+  return (
+    <>
+      <h1 className="title">People Page</h1>
+
+      <div className="block">
+        <div className="box table-container">
+          {isLoading && (
+            <Loader />
+          )}
+
+          {isError && (
+            <p data-cy="peopleLoadingError" className="has-text-danger">
+              Something went wrong
+            </p>
+          )}
+
+          {isNoPeopleOnServer && (
+            <p data-cy="noPeopleMessage">
+              There are no people on the server
+            </p>
+          )}
+
+          {isPeopleOnServer && (
+            <table
+              data-cy="peopleTable"
+              className="table is-striped is-hoverable is-narrow is-fullwidth"
+            >
+              <thead>
+                <tr>
+                  <th>Name</th>
```

Comment:
You can create array with names of columns and map it here

Reply:
➕